soaktest/soakclient/soakclient.go

```diff
@@ -0,0 +1,357 @@
+
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
+
+	select {
+	case sig := <-sigChan:
+		soaktest.InfoLogger.Printf("Signal caught: %v\n", sig)
+	case err := <-errorChan:
+		soaktest.ErrorLogger.Printf("Error caught: %s\n", err)
+	}
+
+	close(doneChan)
+	close(errorChan)
+	for err = range errorChan {
+		soaktest.ErrorLogger.Printf("Error caught: %s\n", err)
+	}
+	wg.Wait()
+	soaktest.PrintConsumerStatus()
+	soaktest.PrintProducerStatus()
+}
+
+// CreateTopic creates the topics if it doesn't exist
+func CreateTopic(ctx context.Context,
+	topics []kafka.TopicSpecification,
+	broker, ccloudAPIKey, ccloudAPISecret string) error {
+	conf := kafka.ConfigMap{
+		"bootstrap.servers": broker,
+		"sasl.mechanisms":   "PLAIN",
+		"security.protocol": "SASL_SSL",
+		"sasl.username":     ccloudAPIKey,
+		"sasl.password":     ccloudAPISecret}
+	AdminClient, err := kafka.NewAdminClient(&conf)
+	if err != nil {
+		soaktest.ErrorLogger.Printf("Failed to create AdminClient: %s\n", err)
+		return err
+	}
+
+	result, err := AdminClient.CreateTopics(ctx, topics)
+	if err != nil {
+		soaktest.ErrorLogger.Printf("Failed to create new topics: %s\n", err)
+		return err
+	}
+
+	for _, res := range result {
+		switch res.Error.Code() {
+		case kafka.ErrTopicAlreadyExists:
+			soaktest.ErrorLogger.Printf("Failed to create topic %s: %s\n",
+				res.Topic, res.Error)
+
+		case kafka.ErrNoError:
+			soaktest.ErrorLogger.Printf("Succeed to create topic %s\n",
+				res.Topic)
+
+		default:
+			err = fmt.Errorf("Failed to create topic %s: %s\n",
+				res.Topic, res.Error)
+			return err
+		}
+	}
+	return nil
+}
+
+// terminateProducer waits for all messages in the Producer queue to
+// be delivered, closes the producer and pass error to the error channel
+func terminateProducer(p *kafka.Producer, err error, errorChan chan error) {
+	remaining := p.Flush(30)
+	soaktest.InfoLogger.Printf("producer: %d message(s) remaining in queue "+
+		"after flush()\n", remaining)
+	p.Close()
+	errorChan <- err
+}
+
+// terminateConsumer closes consumer and pass error to the error channel
+func terminateConsumer(c *kafka.Consumer, err error, errorChan chan error) {
+	c.Close()
+	errorChan <- err
+}
+
+// producer produces messages to input topic
+func producer(inputTopic, broker, ccloudAPIKey, ccloudAPISecret string,
+	wg *sync.WaitGroup, termChan chan bool, errorChan chan error) {
+	defer wg.Done()
+	p, err := kafka.NewProducer(&kafka.ConfigMap{
+		"bootstrap.servers":      broker,
+		"statistics.interval.ms": 5000,
+		"sasl.mechanisms":        "PLAIN",
+		"security.protocol":      "SASL_SSL",
+		"sasl.username":          ccloudAPIKey,
+		"sasl.password":          ccloudAPISecret})
+	if err != nil {
+		soaktest.ErrorLogger.Printf("Failed to create producer %s\n", err)
+		errorChan <- err
+		return
+	}
+
+	run := true
+	doneChan := make(chan bool)
+	tags := []string{fmt.Sprintf("topic: %s", inputTopic)}
```

Comment:
In datadog-go/statsd examples it looks like the tag should not contain a space, so: topic:%s